clientv3/client.go

```diff
@@ -217,6 +218,10 @@ func (c *Client) dialSetupOpts(endpoint string, dopts ...grpc.DialOption) (opts
 		opts = []grpc.DialOption{grpc.WithTimeout(c.cfg.DialTimeout)}
 	}
 	if c.cfg.DialKeepAliveTime > 0 {
+		// prevent incompatible settings that can lead to connection close from server-side
```

Comment:
It should be OK to let this be misconfigured and let the client have trouble (instead of the panic) since it's opt-in. In the future the etcd server may need <MinTime, and it'd be nicer if older clients could use it without any trouble.